vllm/compilation/decorators.py

```diff
@@ -37,48 +71,46 @@ def __init__(self, *args, **kwargs):
 
     cls.__init__ = __init__
 
-    def __call__(
-        self,
-        input_ids: Optional[torch.Tensor],
-        positions: torch.Tensor,
-        kv_caches: List[torch.Tensor],
-        attn_metadata: AttentionMetadata,
-        intermediate_tensors: Optional[IntermediateTensors],
-        inputs_embeds: Optional[torch.Tensor] = None,
-    ) -> Union[torch.Tensor, IntermediateTensors]:
+    def __call__(self, *args, **kwargs):
         # torch.compiler.is_compiling() means we are inside the compilation
         # e.g. TPU has the compilation logic in model runner, so we don't
         # need to compile the model inside.
         if torch.compiler.is_compiling():
-            return self.forward(input_ids, positions, kv_caches, attn_metadata,
-                                intermediate_tensors, inputs_embeds)
+            return self.forward(*args, **kwargs)
 
         # the first compilation needs to have dynamic shapes marked
         if len(self.compiled_codes) < 1:
-            if input_ids is not None:
-                torch._dynamo.mark_dynamic(input_ids, 0)
-            torch._dynamo.mark_dynamic(positions, 0)
-            if inputs_embeds is not None:
-                torch._dynamo.mark_dynamic(inputs_embeds, 0)
-            if intermediate_tensors is not None:
-                for tensors in intermediate_tensors.tensors.values():
-                    torch._dynamo.mark_dynamic(tensors, 0)
+            sig = inspect.signature(self.__class__.forward)
```

Comment:
Should you use bound_args here? I know the refer to the same arguments for passed args, but for defaults, the dimensions won't get marked as dynamic in this implementation, if I understand this correctly

Reply:
I don't get it. we have `bound_args.apply_defaults()` right after this line.

in addition, we should not have `torch.Tensor` as a default value. defaults should not require dynamic annotation.

Reply:
Yeah but where do bound_args get used accept for argument lookup? The forward method is still called with args

Reply:
> The forward method is still called with args

python interpreter will deal with the default. `bound_args` are the arguments **after function call, inside the function**. this is to have a unified way to get, say, `input_ids`, which can be passed via either position or keyword args.